d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -43,16 +43,27 @@ interface ResourceWatcher
   interface NodeResourceWatcher extends ResourceWatcher
   {
     void onChanged(NodeUpdate update);
+    // OnDelete method for interface functional completeness
+    // By now, OnDelete method should not be triggerred, as we don't want to delete the data from the cache which is just a design choice
```

Comment:
Can you break up this long line, and make it a proper javadoc?

Also, this isn't really correct. Ideally, we _do_ want to call `onDelete`, it's just that we will explicitly ignore it. We still want to call it to ensure that the callback is invoked quickly with the cached value, instead of timing out after some time

Reply:
Done